src/BootAppManager.ts

```diff
@@ -65,26 +65,32 @@ export class BootAppManager {
                 this._boot_projects.delete(location);
             } else {
                 if (entries && isBootAppClasspath(entries)) {
-                    this._boot_projects.set(location, new BootApp(location, name, entries, STATE_INACTIVE));
+                    const current: BootApp | undefined = this._boot_projects.get(location);
```

Comment:
Here by default the state is `STATE_INACTIVE`, so when classpath of an existing project changes, to preserve current state, here I update the BootApp instance instead of creating a new one.